src/server/scripts/EasternKingdoms/zone_goldshire.cpp

```diff
@@ -0,0 +1,263 @@
+        };
+
+        Trinity::Containers::RandomShuffle(MovePosPositions);
+
+        // first we break formation because children will need to move on their own now
+        for (auto guid : _childrenGUIDs)
+            if (Creature* child = ObjectAccessor::GetCreature(*me, guid))
+                if (child->GetFormation())
+                    child->GetFormation()->RemoveMember(child);
+
+        // Move each child to an random position
+        for (uint32 i = 0; i < _childrenGUIDs.size(); ++i)
+        {
+            if (Creature* children = ObjectAccessor::GetCreature(*me, _childrenGUIDs.at(i)))
+            {
+                children->SetWalk(true);
+                children->GetMotionMaster()->MovePoint(0, MovePosPositions[i], true, MovePosPositions[i].GetOrientation());
+            }
+        }
+        me->SetWalk(true);
+        me->GetMotionMaster()->MovePoint(0, MovePosPositions.back(), true, MovePosPositions.back().GetOrientation());
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        switch (pathId)
+        {
+            case STORMWIND_PATH:
+            {
+                if (waypointId == STORMWIND_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_GOLDSHIRE, 11min);
+                }
+
+                break;
+            }
+            case GOLDSHIRE_PATH:
+            {
+                if (waypointId == GOLDSHIRE_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_WOODS, 15min);
+                }
+                break;
+            }
+            case WOODS_PATH:
+            {
+                if (waypointId == WOODS_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_HOUSE, 6min);
+                    _events.ScheduleEvent(EVENT_WP_START_LISA, 2s);
+                }
+
+                break;
+            }
+            case HOUSE_PATH:
+            {
+                if (waypointId == HOUSE_WAYPOINT)
+                {
+                    // Move childeren at last point 
+                    MoveTheChildren();
+
+                    // After 30 seconds a random sound should play
+                    _events.ScheduleEvent(EVENT_PLAY_SOUNDS, 30s);
+                }
+                break;
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        time_t time = GameTime::GetGameTime();
+        tm localTm;
+        localtime_r(&time, &localTm);
+
+        // Start event at 7 am
+        if ((localTm.tm_hour == 7 && localTm.tm_min == 0 && localTm.tm_sec > 0) && !_started)
+        {
+            // Begin pathing
+            _events.ScheduleEvent(EVENT_BEGIN_EVENT, 2s);
+            _started = true;
```

Comment:
should the code return here ?

Reply:
What do you mean? it will only run once

Reply:
I mean if it should return; when setting _started to true since no event will be executed below in the same UpdateAI call anyway (2s delay for begin event)

Reply:
Don't think it should return here.